### PR TITLE
#CP-22 bug: Fix status code for request forbidden

### PR DESCRIPTION
**Pull request by feyton**
### What the PR does:
Ensure that all status code are matched with the standard status codes in regards to **Token verification** and **Access control**

### Background tasks to be completed

- [x] Fix the status code to match `403` for **Permission denied**
- [x] Fix the issue in `AuthVerify.js` that use `res.send` instead of `res.json`
### How should this be manually tested
Nothing changed in regards to environment setup if you have the project running.
Follow the specs for quick get started:
```js
git clone https://github.com/atlp-rwanda/phantom-be-codebandits
cd phantom-be-codebandits
touch .env // place the entries in .env.sample into this file
npm i
npm run dev
// remember to run your database
// Use POSTMAN or Swagger documentation by going to http://localhost:5000 <Replace "5000" with your "PORT">
```
### Background context
Prior to these changes we were using `401` status for both `Invalid token and missing permission` according to **[Mozilla Documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses)** on `Status code`

> **401** code refer to missing token or invalid token. The server fail to identify the user.

> **403** code refer to missing required permissions. The server is able to identify the user, but the user misses the required permission. Which refers to the use of **Permission denied** status code in such instances

### Relevant Trello cards

<blockquote class="trello-card"><a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;pciaziF2&#x2F;22-status-code-in-response">Status code in response</a></blockquote>

<footer>

[Finishes #CP-22]

### CODE REVIEW

**Review by Yvad60 — DISMISSED** [review]
Good work Fab I noticed the issue that with the response is now fixed in the check permission function and the tests are well updated if it is possible to fix some of codeclimate issues without sacrificing much on the code understandability you may look into that but for now I'm approving your changes

**Review by Abdulkeza — DISMISSED** [review]
Great job Mr fab, I checked all of the endpoints we have and I noticed that the status code is perfect as expected especially from 400<500 which indicates an error caused by the user like a bad request or permissions. so this is a good job.

### NB:  there is an issue that I noticed in registering a new user. I am able to register a user with "Blablabla" as a role value which is not good, like a user shouldn't have a role of kaka.     sorry for this coz it is not related but if possible you can fix it and goes with what you did.

**Review by samyshyaka — DISMISSED** [review]
Nice work @feyton, I think the PR is ready to be merged

**Review by renepromesse — DISMISSED** [review]
Hey @feyton , nice job, would you look into codeclimate issue before we merge this?

**Comment by feyton** [comment]
> Hey @feyton , nice job, would you look into codeclimate issue before we merge this?

The issue with codeclimate in regard to those functions can not be fixed by refactoring as they are related to the AccessControl we are using

File: src/operatorApp/tests/operator.test.js [hunk 1]
```diff
@@ -180,4 +180,140 @@ describe('Operator router tests', () => {
 			.set('Authorization', `Bearer ${token}`);
 		expect(response).to.have.status(400);
 	});
+	// Testing for logged in as an operator
+	it('should login an operator user', async () => {
+		const password = await bcrypt.hash('Andela@1234', 5);
+		const newUser = User.create({
```
Comment (renepromesse) [inline src/operatorApp/tests/operator.test.js]: hey @feyton , could you create a separate file and put in these testing dummy data?

File: src/operatorApp/tests/operator.test.js [hunk 2]
```diff
@@ -180,4 +180,140 @@ describe('Operator router tests', () => {
...
+		const info = { email: user.email, password: 'Andela@1234' };
+		const response = await chai
+			.request(app)
+			.post('/api/v1/accounts/login')
+			.send(info);
+		expect(response).to.have.status(200);
+		token = response.body.data.access_token;
+	});
+	it('should not create an operator', async () => {
+		operatorInfo = {
```
Comment (renepromesse) [inline src/operatorApp/tests/operator.test.js]: move  in that file this too

File: src/operatorApp/tests/operator.test.js [hunk 3]
```diff
@@ -180,4 +180,140 @@ describe('Operator router tests', () => {
...
+		};
+		const response = await chai
+			.request(app)
+			.post('/api/v1/operators')
+			.send(operatorInfo)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(403);
+	});
+	it('should not create an operator', async () => {
+		operatorInfo = {
```
Comment (renepromesse) [inline src/operatorApp/tests/operator.test.js]: create an object in that file and call it everywhere you're using it

File: src/operatorApp/tests/operator.test.js [hunk 4]
```diff
@@ -180,4 +180,140 @@ describe('Operator router tests', () => {
...
+		const info = { email: user.email, password: 'Andela@1234' };
+		const response = await chai
+			.request(app)
+			.post('/api/v1/accounts/login')
+			.send(info);
+		expect(response).to.have.status(200);
+		token = response.body.data.access_token;
+	});
+	it('should not create an operator', async () => {
+		operatorInfo = {
```
Comment (renepromesse) [inline src/operatorApp/tests/operator.test.js]: do the sam on this object too

File: src/operatorApp/tests/operator.test.js [hunk 5]
```diff
@@ -180,4 +180,140 @@ describe('Operator router tests', () => {
...
+		};
+		const response = await chai
+			.request(app)
+			.post('/api/v1/operators')
+			.send(operatorInfo)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(403);
+	});
+	it('should not update an operator', async () => {
+		operatorInfo = {
```
Comment (renepromesse) [inline src/operatorApp/tests/operator.test.js]: and here. look everywhere it should be applied and apply it

**Review by renepromesse — CHANGES_REQUESTED** [review]
@feyton  could you please quick fix these feedback and let me know?

File: src/operatorApp/tests/operator.test.js (hunk 2 above)
Comment (feyton) [reply src/operatorApp/tests/operator.test.js]: These are not tests related to this PR actually, they are tests for CRUD operations for bus. I changed these when I edited the expected response from 401 to 403

File: src/operatorApp/tests/operator.test.js (hunk 3 above)
Comment (feyton) [reply src/operatorApp/tests/operator.test.js]: I am working on those fixes right now

File: src/driverApp/tests/driver.test.js [hunk 6]
```diff
@@ -21,14 +21,14 @@ describe('driver router tests', () => {
 	let token;
 	let createdDriverId;
 	let driverInfo;
-	it('should login an admin user', async () => {
+	it('should login an operator user', async () => {
 		const password = await bcrypt.hash('Andela@1234', 5);
```
Comment (renepromesse) [inline src/driverApp/tests/driver.test.js]: use the password created in dummy data file

File: src/driverApp/tests/driver.test.js [hunk 7]
```diff
@@ -21,14 +21,14 @@ describe('driver router tests', () => {
...
 		const newUser = User.create({
 			email: 'test123@gmail.com',
 			password,
 			firstName: 'test',
 			lastName: 'Andela',
-			role: 'admin',
+			role: 'operator',
```
Comment (renepromesse) [inline src/driverApp/tests/driver.test.js]: use the operator you created in dummy data instead of creating a new user here

**Review by renepromesse — DISMISSED** [review]
please fix this as well

File: src/driverApp/tests/driver.test.js (hunk 6 above)
Comment (feyton) [reply src/driverApp/tests/driver.test.js]: I just fixed that issue in the latest push. Take a look

File: src/driverApp/tests/driver.test.js (hunk 7 above)
Comment (feyton) [reply src/driverApp/tests/driver.test.js]: That one too is resolved

**Review by renepromesse — APPROVED** [review]

**Review by Yvad60 — APPROVED** [review]

**Review by Abdulkeza — APPROVED** [review]

**Pull request merged**